statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,721 @@
+                                                 Poisson, Logit, CountResults,
+                                                 L1CountResults, Probit,
+                                                 _discrete_results_docs,
+                                                 GeneralizedPoisson,
+                                                 NegativeBinomialP)
+from statsmodels.distributions import zipoisson, zigenpoisson, zinegbin
+from statsmodels.tools.numdiff import (approx_fprime, approx_hess,
+                                       approx_hess_cs, approx_fprime_cs)
+from statsmodels.tools.decorators import (resettable_cache, cache_readonly)
+
+
+class GenericZeroInflated(CountModel):
+    __doc__ = """
+    Generiz Zero Inflated model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    exog_infl: array
+        A reference to the zero-inflated exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, exog_infl=None, offset=None,
+                 inflation='logit', exposure=None, missing='none', **kwargs):
+        super(GenericZeroInflated, self).__init__(endog, exog, offset=offset,
+                                                  exposure=exposure,
+                                                  missing=missing, **kwargs)
+
+        if exog_infl is None:
+            self.k_inflate = 1
+            self.exog_infl = np.ones((endog.size, self.k_inflate))
+        else:
+            self.exog_infl = exog_infl
+            self.k_inflate = exog_infl.shape[1]
+
+        if len(exog.shape) == 1:
+            self.k_exog = 1
+        else:
+            self.k_exog = exog.shape[1]
+
+        self.infl = inflation
+        if inflation == 'logit':
+            self.model_infl = Logit(np.zeros(self.exog_infl.shape[0]),
+                                    self.exog_infl)
+            self._hessian_inflate = self._hessian_logit
+        elif inflation == 'probit':
+            self.model_infl = Probit(np.zeros(self.exog_infl.shape[0]),
+                                    self.exog_infl)
+            self._hessian_inflate = self._hessian_probit
+
+        else:
+            raise TypeError("inflation == %s, which is not handled"
+                % inflation)
+
+        self.inflation = inflation
+        self.k_extra = self.k_inflate
+        self.exog_names.insert(0, 'inflate_const')
+        for i in range(self.k_extra - 1, 0, -1):
+            self.exog_names.insert(0, 'inflate_x%d' % i)
```

Comment:
these are default names and it should use the provided names if pandas was used

Reply:
```
-----------------------------------------------------------------------------------------
inflate_constant         -1.3512      0.890     -1.519      0.129      -3.095       0.392
inflate_sex              -0.2878      0.240     -1.201      0.230      -0.757       0.182
inflate_age               0.0440      0.018      2.410      0.016       0.008       0.080
inflate_nmarried         -0.0922      0.032     -2.853      0.004      -0.156      -0.029
inflate_nchildren        -0.4103      0.293     -1.403      0.161      -0.984       0.163
inflate_religious         0.3184      0.090      3.536      0.000       0.142       0.495
inflate_education        -0.0224      0.051     -0.441      0.659      -0.122       0.077
inflate_occupation       -0.0280      0.072     -0.389      0.697      -0.169       0.113
inflate_rate_marriage     0.4644      0.091      5.097      0.000       0.286       0.643
constant                  2.5574      0.286      8.934      0.000       1.996       3.118
sex                      -0.1195      0.080     -1.487      0.137      -0.277       0.038
age                      -0.0038      0.006     -0.592      0.554      -0.016       0.009
nmarried                  0.0489      0.012      4.252      0.000       0.026       0.071
nchildren                -0.1904      0.098     -1.950      0.051      -0.382       0.001
religious                -0.1282      0.031     -4.081      0.000      -0.190      -0.067
education                -0.0288      0.018     -1.618      0.106      -0.064       0.006
occupation                0.0555      0.026      2.115      0.034       0.004       0.107
rate_marriage            -0.1088      0.029     -3.703      0.000      -0.166      -0.051
=========================================================================================
```
Are these names ok?